semantic/src/main/kotlin/tools/samt/semantic/checks/ValidTypeExpressionCheck.kt

```diff
@@ -0,0 +1,155 @@
+package tools.samt.semantic.checks
+
+import tools.samt.common.DiagnosticConsole
+import tools.samt.parser.*
+import tools.samt.semantic.SemanticCheck
+
+internal class ValidTypeExpressionCheck(private val diagnostics: DiagnosticConsole) : SemanticCheck {
+    override fun check(fileNode: FileNode) {
+        fileNode.statements.forEach {
+            when (it) {
+                is RecordDeclarationNode -> it.fields.forEach { field -> checkTypeExpression(field.type) }
+                is TypeAliasNode -> checkTypeExpression(it.type)
+                is TypeImportNode,
+                is WildcardImportNode,
+                is EnumDeclarationNode,
+                is ConsumerDeclarationNode,
+                is PackageDeclarationNode,
+                is ProviderDeclarationNode,
+                is ServiceDeclarationNode,
```

Comment:
`ServiceDeclarationNode` can also contain type expressions inside the operation parameter lists.

Reply:
You are very correct, I totally forgot those parameters 🙈

Reply:
I now also check the parameter types and return type